compare_experiments/nbconverted/compare_generic_genes.py

```diff
@@ -0,0 +1,210 @@
+template_filename2 = os.path.join(data_dir, "data", f"processed_recount2_template_{project_id2}.tsv")
+
+
+# ## Correlation between rankings
+
+# In[4]:
+
+
+# Load gene ranking
+gene_ranking_summary1 = pd.read_csv(gene_ranking_filename1, sep="\t", index_col=0, header=0)
+gene_ranking_summary2 = pd.read_csv(gene_ranking_filename2, sep="\t", index_col=0, header=0)
+
+
+# In[5]:
+
+
+# Get simulated ranking
+gene_ranking1 = gene_ranking_summary1["Rank (simulated)"].rename("Rank 1")
+gene_ranking2 = gene_ranking_summary2["Rank (simulated)"].rename("Rank 2")
+
+# Combine ranking
+gene_ranking_combined = pd.concat([gene_ranking1, gene_ranking2], axis=1)
+
+
+# In[6]:
+
+
+print(gene_ranking_combined.shape)
+gene_ranking_combined.head()
+
+
+# In[7]:
+
+
+# Check for NAs
+gene_ranking_combined[pd.isnull(gene_ranking_combined).any(axis=1)]
+
+
+# In[8]:
+
+
+# Plot correlation between ranking
+fig = sns.jointplot(
+    data=gene_ranking_combined,
+    x="Rank 1",
+    y="Rank 2",
+    kind="hex",
+    marginal_kws={"color": "white"},
+)
+
+fig.set_axis_labels(
+    f"Ranking in {project_id1}", f"Ranking in {project_id2}", fontsize=14, fontname="Verdana"
+)
+
+output_figure_filename = "concordance_between_recount2_templates.svg"
+fig.savefig(
+    output_figure_filename,
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+
+
+# **Takeaway:**
+# 
+# * Looks like there is good concordance between highly ranked genes (i.e. generic genes)
+# * In general, we expect that there are some genes that are generally generic (i.e. those that are concordant between these two experiments) and there are also genes that are generic within the given context of the specific experiment.
+
+# ## Examine gene expression data
+
+# In[9]:
+
+
+# Read expression data
+template_1 = pd.read_csv(template_filename1, sep="\t", index_col=0, header=0)
+template_2 = pd.read_csv(template_filename2, sep="\t", index_col=0, header=0)
+
+
+# In[10]:
+
+
+# Get concordance genes
+concordant_genes = list(gene_ranking_combined[(gene_ranking_combined["Rank 1"]>15000) &
+                                         (gene_ranking_combined["Rank 2"]>15000)].index)
+
+# Get disconcordant genes
+discordant_genes = set(gene_ranking_combined.index).difference(concordant_genes)
+
+
+# In[11]:
+
+
+# Distribution of concordant genes in template experiment 1
+template1_mean = template_1.mean()
+
+print("Percent concordant genes with 0 expression in template 1:", 
+      len(template1_mean[concordant_genes].loc[template1_mean[concordant_genes]==0])/len(template1_mean[concordant_genes]))
+
+print("Percent nonzero concordant genes in template 1:", 
+      len(template1_mean[concordant_genes].loc[(template1_mean[concordant_genes]>0) &
+                                           (template1_mean[concordant_genes]<1000)])/len(template1_mean[concordant_genes]))
+sns.distplot(template_1.mean()[concordant_genes], kde=False)
```

Comment:
Could you add axis labels here? It's not quite clear what you're plotting at first glance (although I figured it out by reading the code/comments).

Same comment for the next few plots.